### PR TITLE
Feature/unblock

### PR DESCRIPTION
**Pull request by louiscrc**
# New Feature : unblock Profiles
## UI
The UI is quite similar to the AroundYou page, except it displays user cards of users you blocked rather than regular users. 

Also, the TopBar is thinner as it's not a BottomNavigation page. And of course there is no Filter button.

Finally, new button in the Settings to replace the old `Unblock All` button, which gives access to the UnblockProfile screen.

## Back-end
Few changes in the back-end:

- ``getBlockedProfiles``: new function in the ProfileViewModel that triggers the same function in ProfilesRepositoryFirebase to get the Profiles of the users in `hasBlocked` from the database.
- ``unblockUser``: new function in the ProfileViewModel that deletes removes a the uid parameter from the hasBlocked list of the selfProfile.

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt [hunk 1]
```diff
@@ -143,6 +143,15 @@ fun SettingsScreen(
                 Text("Unblock All", color = Color.White)
               }
 
+        Button(
+            onClick = {
+                navigationActions.navigateTo(Screen.UNBLOCK_PROFILE)
+            },
+            colors = ButtonDefaults.buttonColors(containerColor = Color.Blue),
+            modifier = Modifier.fillMaxWidth()) {
+            Text("See unblock", color = Color.White)
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt:152]: Nitpick: Hardcoded string could be put in Strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+          onDismissRequest = { profileToUnblock = null },
+          title = { Text("Confirm Unblock") },
+          text = { Text("Are you sure you want to unblock ${profileToUnblock!!.name}?") },
+          confirmButton = {
+              TextButton(onClick = {
+                  profilesViewModel.unblockUser(profileToUnblock!!.uid)
+                  profilesViewModel.getBlockedUsers()
+                  profileToUnblock = null
+              }) {
+                  Text("Yes")
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:94]: Nitpick: Hardcoded string could be put in Strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+              }
+          },
+          dismissButton = {
+              TextButton(onClick = { profileToUnblock = null }) {
+                  Text("No")
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:99]: Nitpick: Hardcoded string could be put in Strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+  LaunchedEffect(isConnected.value) {
+    if (!isTestMode && !isNetworkAvailable()) {
+      profilesViewModel.updateIsConnected(false)
+    } else {
+      while (true) {
+        // Call the profile fetch function
+        profilesViewModel.getBlockedUsers()
+
+        // Wait 10 seconds before the next update
+        delay(10_000L)
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:115]: Nitpick: magic number

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+                      navigationActions.navigateTo(route.route)
+                  }
+              },
+              tabList = LIST_TOP_LEVEL_DESTINATIONS,
+              selectedItem = Route.UNBLOCK_PROFILE)
+      },
+      topBar = {
+          TopAppBar(
+              modifier = Modifier.testTag("unblockTopBar"),
+              title = { Text("Blocked Users",color= Color.White, modifier = Modifier.testTag("UnblockTopBarTitle")) },
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:138]: Nitpick: Hardcoded string could be put in Strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+                  }
+                  else {
+                      navigationActions.navigateTo(route.route)
+                  }
+              },
+              tabList = LIST_TOP_LEVEL_DESTINATIONS,
+              selectedItem = Route.UNBLOCK_PROFILE)
+      },
+      topBar = {
+          TopAppBar(
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:136]: You could use the standardized topBar we have in Ui/Shared

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+                  verticalArrangement = Arrangement.spacedBy(COLUMN_VERTICAL_PADDING),
+                  modifier =
+                      Modifier.fillMaxSize().padding(horizontal = COLUMN_HORIZONTAL_PADDING)) {
+                    if (!isConnected.value) {
+                      item {
+                        Box(
+                            contentAlignment = Alignment.Center,
+                            modifier = Modifier.fillMaxSize().testTag("noConnectionPrompt")) {
+                              Text(
+                                  text = "No Internet Connection",
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:170]: Nitpick: Hardcoded string could be put in Strings.xml or there might already be one for this I think.

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+  val isConnected = profilesViewModel.isConnected.collectAsState()
+
+  var profileToUnblock by remember { mutableStateOf<Profile?>(null) }
+
+  if (profileToUnblock != null) {
+      AlertDialog(
+          modifier = Modifier.testTag("unblockDialog"),
+          onDismissRequest = { profileToUnblock = null },
+          title = { Text("Confirm Unblock") },
+          text = { Text("Are you sure you want to unblock ${profileToUnblock!!.name}?") },
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:87]: Nitpick: Hardcoded string could be put in Strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+  val context = LocalContext.current
+  val isConnected = profilesViewModel.isConnected.collectAsState()
+
+  var profileToUnblock by remember { mutableStateOf<Profile?>(null) }
+
+  if (profileToUnblock != null) {
+      AlertDialog(
+          modifier = Modifier.testTag("unblockDialog"),
+          onDismissRequest = { profileToUnblock = null },
+          title = { Text("Confirm Unblock") },
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt:86]: Nitpick: Hardcoded string could be put in Strings.xml

**Review by BotondAKovacs — CHANGES_REQUESTED** [review]
Good PR overall, it works perfectly, there are however some nitpicks that can easily be fixed, like hardcoded strings and numbers and also a suggestion to use a shared ui component.
Tests are lacking for now.
Question: I'm wondering what's better between not letting offline users see the blocklist and what you've done, which is letting them go there and display that the user is not connected. Maybe just putting a toast when trying to go on that screen makes more sense. What do you think?

The topbar is incoherent with the rest of the app but my suggestion should fix that. 
Questions: Is the bottom navigation bar needed? 
Screenshot 2024-11-21 at 18 21 53

**Comment by louiscrc** [comment]
Though I agree with you for the TopBar component, this needs to be stated as an issue and fixed later. The current TopBar doesn't allow a goBack button...

**Comment by BotondAKovacs** [comment]
> Though I agree with you for the TopBar component, this needs to be stated as an issue and fixed later. The current TopBar doesn't allow a goBack button...

I pushed the new one to the main yesterday

**Comment by louiscrc** [comment]
> > Though I agree with you for the TopBar component, this needs to be stated as an issue and fixed later. The current TopBar doesn't allow a goBack button...
> 
> I pushed the new one to the main yesterday

Indeed, I merged the main into my branch and saw it. I'll implement it.

**Comment by BotondAKovacs** [comment]
The dialog seems broken in the last version
image

**Review by BotondAKovacs — APPROVED** [review]
You should link this PR to a scrumboard issue and the coverage could be improved in the future, apart from that LGTM

**Review by BotondAKovacs — APPROVED** [review]
LGTM

**Pull request merged**